### PR TITLE
Private Events / Microverse Association Project

### PR DESCRIPTION
**Pull request by gurbuzertunga**
In this project we;

- Implemented a User, Events and Event Attendees databases and relationship between them,
- Started Using Tailwind CSS as a practice and CSS styling of the project,
- A User can create and join different events,
- An Event can have multiple different attendees and these will show in Event's show page,
- A user can attend to multiple events and these will show in User's show page.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -6,10 +6,34 @@
...
+              <%= link_to "Log out", destroy_user_session_path, method: :delete, class: "text-gray-100 text-lg mx-3 capitalize" %>
+            </p>
+          <% else %>
+	          <p>
+              <%= link_to 'Sign Up', new_user_registration_path, class: "text-gray-100 text-lg mx-3 capitalize" %>
+            </p>
+            <p>
+              <%= link_to 'Sign In', new_user_session_path, class: "text-gray-100 text-lg mx-3 capitalize" %>
+            </p>
+          <% end %>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb:33]: Kindly remove logic from views. Move them into view helpers.

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,14 @@
 Rails.application.routes.draw do
+  get 'events/new'
+  get 'events/index'
   devise_for :users, :controllers => { registrations: 'registrations'}
   resources :users
+  resources :events
```
Comment (maelfosso) [inline config/routes.rb]: **`OPTIONAL`**
Routes can be restricted to this `resources :events, only: [:new, :create, :show, :index]`

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,9 +1,45 @@
...
+                    <p><%= event.date %></p><br/>
+                    <strong>Description :</strong>
+                    <p><%= event.description %></p><br/>
+    <% end %>
+
+    <strong>Attending Following Events:</strong><br/>
+  <% @user.event_attended.each do |t| %>
+
+    <%= t.description %> </br>
+  <% end %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb]: That block is not really necessary.
You can remove it.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# changes required :red_circle:

Hello,

Thank you the well done job so far
However, there are some issues to fix 

1. Kindly add **indexes** in `event_attendees` table for the two references

2. Kindly add unit and integrations as indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) and [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png)

3- It's possible, in your app, to create an event without being logged in
Screenshot at 2020-10-14 16:45:21
Read the requirement [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_can_create_events_requirement.png). Kindly fix that, please.
To fix that use `before_action` for all the actions into EventsController

4- If you take a closely look at this screenshot
Screenshot at 2020-10-14 16:51:39
You will notice that **alice** appears twice in the list of attendees.
Kindly fix that, please.

5- There is the screenshot of users' list
Screenshot at 2020-10-14 16:52:55
Kindly improve it because it's difficult to understand. Use a `<table>` for example

6- Read the comments left below

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi team! Great job implementing most of the suggested changes in the last review. There are a few issues I'd like you to consider before moving forward though. These include:
- [x] The `New User` button on your users' page doesn't work as expected. It returns an error. Either fix it or remove it so that the new user can only be created from the `Sign Up` link in the navbar.
- [x] __[Optional]__ Your authentication system lacks some vital functionality e.g. when an unlogged-in user tries to register for an event, an error is returned. Please fix your authentication so that the user is shown an appropriate message. Probably indicating that one has to be logged in to register for an event. This is the same for the edit action of any event.
- [x] Add integration tests for authentication (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [x] Add integration tests event managment features (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/)) __[Hint]__ [Here](https://www.codewithjason.com/rails-integration-tests-rspec-capybara/) is a nice article about writing integration tests.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Review by trekab — APPROVED** [review]
### Approved! :tada: 
Well done implementing the suggested changes, team! Keep up the good work!

_Happy coding!_
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Pull request merged**